Makefile

```diff
@@ -46,6 +46,14 @@ DEPS = $(LLVM_EXT_OBJ)
 CXXFLAGS += $(if $(debug),-g -O0)
 CRYSTAL_VERSION ?= $(shell cat src/VERSION)
 
+DESTDIR ?=
+PREFIX ?= /usr/local
```

Comment:
What's the rationale for having `DESTDIR` in addition to `PREFIX`?

Reply:
We're already using the same setup in `shards`. And it seems to be commonly used (https://stackoverflow.com/questions/11307465/destdir-and-prefix-of-make).
The original purpose is probably that the path `PREFIX` could be baked into the binary, so it's set at `./configure`. And `DESTDIR` is only used for the `install` recipe directly. We wouldn't really need that with #10877 and can get the same functionality with `PREFIX` only (which can include the path to `DESTDIR`). But I think it still makes sense to have both.